src/main/java/org/opensearch/security/privileges/PitPrivilegesEvaluator.java

```diff
@@ -0,0 +1,356 @@
+                                        IndexNameExpressionResolver resolver,
+                                        PrivilegesEvaluatorResponse presponse) {
+        Map<String, String[]> pitToIndicesMap = OpenSearchSecurityPlugin.
+                GuiceHolder.getPitService().getIndicesForPits(pitIds);
+        Set<String> pitIndices = new HashSet<>();
+        // add indices across all PITs to a set and evaluate if user has access to all indices
+        for(String[] indices: pitToIndicesMap.values()) {
+            pitIndices.addAll(Arrays.asList(indices));
+        }
+        Set<String> allPermittedIndices = getPermittedIndices(pitIndices, clusterService, user,
+                securityRoles, action, resolver);
+        // In this case, PIT IDs are explicitly passed.
+        // So, only if user has access to all PIT's indices, allow delete operation, otherwise fail.
+        if(pitIndices.size() == allPermittedIndices.size()) {
+            presponse.allowed = true;
+            presponse.markComplete();
+        }
+        return presponse;
+    }
+
+    /**
+     * This method returns list of permitted indices for the PIT indices passed
+     */
+    private Set<String> getPermittedIndices(Set<String> pitIndices, ClusterService clusterService,
+                                             User user, SecurityRoles securityRoles, final String action,
+                                             IndexNameExpressionResolver resolver) {
+        final ImmutableSet<String> INDICES_SET = ImmutableSet.copyOf(pitIndices);
+        final IndexResolverReplacer.Resolved pitResolved =
+                new IndexResolverReplacer.Resolved(INDICES_SET, INDICES_SET, INDICES_SET,
+                        ImmutableSet.of(), SearchRequest.DEFAULT_INDICES_OPTIONS);
+        return securityRoles.reduce(pitResolved,
+                user, new String[]{action}, resolver, clusterService);
+    }
+
+    /**
+     * Get all active PITs
+     */
+    private List<ListPitInfo> getAllPitInfos(GetAllPitNodesRequest request) throws InterruptedException {
+        final List<ListPitInfo> pitInfos = new ArrayList<>();
+        final CountDownLatch latch = new CountDownLatch(1);
+        ActionListener listener = new ActionListener<GetAllPitNodesResponse>() {
+            @Override
+            public void onResponse(GetAllPitNodesResponse response) {
+                pitInfos.addAll(response.getPitInfos());
+                request.setGetAllPitNodesResponse(response);
+            }
+
+            @Override
+            public void onFailure(Exception e) {
+                throw new OpenSearchException("List all PITs failed", e);
+            }
+        };
+        LatchedActionListener latchedActionListener = new LatchedActionListener<>(listener, latch);
+
+        OpenSearchSecurityPlugin.GuiceHolder.getPitService().getAllPits(latchedActionListener);
+
+        if(!latch.await(15, TimeUnit.SECONDS)) {
```

Comment:
@reta as per this [discussion](https://github.com/opensearch-project/OpenSearch/pull/4064#discussion_r963063762) we need to get rid of this latch since this has the potential to block http request handler threads. 

Do you have thoughts on how we could have a reference to the response listener or perform non-blocking waits,

Reply:
Sorry @Bukhtawar , I came a bit late, seems like this code is already gone :partying_face:

Reply:
@bharath-techie would you be able to shed some light if you still have that commit handy

@reta apologies let's see if we that commit